compiler/rustc_lint/src/lints.rs

```diff
@@ -613,11 +613,19 @@ pub struct ExpectationNote {
     pub rationale: Symbol,
 }
 
-// fn_null_check.rs
+// ptr_nulls.rs
 #[derive(LintDiagnostic)]
-#[diag(lint_fn_null_check)]
-#[help]
-pub struct FnNullCheckDiag;
+pub enum PtrNullChecksDiag<'a> {
+    #[diag(lint_ptr_null_checks_fn_ptr)]
+    #[help(lint_help)]
+    FnPtr,
```

Comment:
Should the expression's span and type be added here too, for symmetry?

Reply:
Don't know, but added anyway since it doesn't make the output any worse.